improver_tests/acceptance/SHA256SUMS

```diff
@@ -188,6 +188,14 @@ c2abfc4f83d4a918dd2d80f20b7746e5ae19d809d21df9715bd8ed5011d38ed2  ./extract/site
 f3516e5b36fa637151c4c7306e08d348fdc04aa24dec07e2bcd333248f4010cc  ./field-texture/basic/kgo.nc
 3379e6f19083154ebea42a32d9b48e0bae442243833aed23e45dea61b24d7739  ./fill-radar-holes/basic/201811271330_remasked_rainrate_composite.nc
 b25a397ac5ecac536f43c553f17b0bd6b5d2587d4eda8559bf5709e25f2c237b  ./fill-radar-holes/basic/kgo.nc
+16fa127db4338362b87dda00270c4490d4e501f7dff0cd8e4193bcee55cab727  ./freezing-rain/one_hour/kgo.nc
+0ddfec929063042e7dc47d0ef154f539a1d7418f74a724343ae42651e714d6b2  ./freezing-rain/one_hour/rain_acc.nc
+c7258e05788b7605524be7f0074c81a71ebd98683dc5b24e5bad0ac1703e74b9  ./freezing-rain/one_hour/sleet_acc.nc
+44c14d083e3e1154e9a87ce4c3e6921dc81816b8bba9dc1da40c16b7ef612cf9  ./freezing-rain/one_hour/temperature_min.nc
+b3c234a150d259e078ca9c42e49458619c5630212f64b6d7c4ec2e94cd65c87f  ./freezing-rain/three_hour/kgo.nc
+ecd65249d525dec4662e6047363423646f59bad26aa3b13e068f1d94c028bcfe  ./freezing-rain/three_hour/rain_acc.nc
+e326b32d7382c9a2244feaae19025234bd080ec3252888adfe1c83d217703ab2  ./freezing-rain/three_hour/sleet_acc.nc
+ecf9b9ca7c96725e2304b5bcb98f9e0ac832cc16e4110cca94189357cf303a8c  ./freezing-rain/three_hour/temperature_min.nc
```

Comment:
realization coords don't match!
I can see that you have tried to provide one extra temperature realization than are present for precipitation realizations to mimic how we plan to run this, but the realization IDs don't match. This suggests that something isn't right in the Plugin.

| Period | Precip realization IDs | Temperature realization IDs |
|--------|------------------------|-----------------------------|
| 1H     | [3, 4, 5]              | [0, 1, 2, 3]                |
| 3H     | [0, 18, 19, 20, 21]    | [0, 1, 2, 3, 4, 5]          |

Reply:
Thanks for spotting this Stephen, it highlighted a fairly fundamental error in the plugin. It was using a reference cube (the cube with the smallest number of realizations) and ensuring that only realizations that matched the numbers in that reference cube were kept. However, it did not compare all of the cubes against each other, meaning:
```
Inputs
ref   : 0, 1, 2
cube 1: 0, 1, 2, 3
cube 2: 0, 4, 5,6

Outputs
ref   : 0, 1, 2
cube 1: 0, 1, 2
cube 2: 0
```
Numpy broadcasting then magically handled the mismatch in the number of realizations, which was not dreadfully helpful. I've modified this to create a set of common realizations across the three input cubes. It will also now raise an exception if there are no realizations in common; I've not imposed a minimum number.

This PR now also includes the plugin update, modified unit tests that would have flagged the original method as flawed, and an additional test for the new exception.